pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -151,6 +151,8 @@
     private CompletableFuture<Long> currentCompaction = CompletableFuture.completedFuture(COMPACTION_NEVER_RUN);
     private final CompactedTopic compactedTopic;
 
+    private final boolean isSystemTopic;
```

Comment:
see my comment above about making a new `SystemTopic` class and making it extend `PersistentTopic`.